### PR TITLE
PHP 7.1 & Symfony 4

### PR DESCRIPTION
**Pull request by zhem4ag**
Updated project to support PHP 7.1 and Symfony 4

### CODE REVIEW

**Comment by blaugueux** [comment]
@zhem4ag why dropping php 5.6 support?

**Comment by markitosgv** [comment]
I think we have to support from PHP 5.6 to 7.1. Or maybe create a new release 1.x supporting only Symfony >3.4 and PHP >7.1.

What do you think?

**Comment by zhem4ag** [comment]
I think best solutions will be - create new release for latest Symfony version support.
Because starting from Symfony 4, it will require PHP >=7.1

**Comment by blaugueux** [comment]
Right, that should be a v1.0.0 because of BC breaks.

**Comment by markitosgv** [comment]
Please @zhem4ag could you try to fix this PR?

**Comment by xthiago** [comment]
I like the idea of drop support for previous versions of Symfony (<3.4) and PHP (< 7.1). 👍

**Comment by blaugueux** [comment]
@zhem4ag any news on this PR. Do you need help?

**Comment by harold20** [comment]
It's necessary update the project for symfony 4!!

**Comment by zhem4ag** [comment]
I'll update PR on this week.

**Comment by dleute** [comment]
Is there a status on this?

**Comment by zhem4ag** [comment]
Updated. Ready to be merged.

File: Doctrine/RefreshTokenManager.php [hunk 1]
```diff
@@ -23,45 +23,25 @@ class RefreshTokenManager extends BaseRefreshTokenManager
...
-    /**
-     * @param string $username
-     *
-     * @return RefreshTokenInterface
-     */
-    public function getLastFromUsername($username)
+    public function getLastFromUsername(string $username): ? RefreshTokenInterface
     {
-        return $this->repository->findOneBy(array('username' => $username), array('valid' => 'DESC'));
+        return $this->repository->findOneBy(['username' => $username]);
```
Comment (rimvislt) [inline Doctrine/RefreshTokenManager.php]: you will have more than 1 token inside database so 'valid' => 'DESC' here was taking the last token

File: Doctrine/RefreshTokenManager.php [hunk 2]
```diff
@@ -84,12 +60,9 @@ public function delete(RefreshTokenInterface $refreshToken, $andFlush = true)
     }
 
     /**
-     * @param \DateTime $datetime
-     * @param bool      $andFlush
-     *
      * @return RefreshTokenInterface[]
      */
-    public function revokeAllInvalid($datetime = null, $andFlush = true)
+    public function revokeAllInvalid(\DateTime $datetime = null, $andFlush = true): array
```
Comment (rimvislt) [inline Doctrine/RefreshTokenManager.php]: ```use DateTime;``` ?

**Comment by merlingore** [comment]
Any update on this?

**Comment by arm1n** [comment]
Would also like to hear something about ETA? Thanks!

**Comment by markitosgv** [comment]
Already sf4 compatible

**Pull request closed without merging**